lib/wlib/exception/Exception.h

```diff
@@ -0,0 +1,70 @@
+/**
+ * @file Exception.h
+ * @brief Custom exception types for WLib.
+ *
+ * @author Jeff Niu
+ * @date November 2, 2017
+ * @bug No known bugs
+ */
+
+/*
+ * TODO: Use dynamic string type once implemented
+ * TODO: Define RangeException
+ */
+
+#ifndef EMBEDDEDTESTS_EXCEPTION_H
+#define EMBEDDEDTESTS_EXCEPTION_H
+
+#include <stdint.h>
+#include <strings/StaticString.h>
+
+namespace wlp {
+
+    /**
+     * Base exception class from which other exceptions derive.
+     * @tparam tSize size of the static string message
+     */
+    class Exception {
+    public:
+        typedef const char* msg_type;
+
+    protected:
+        Exception() throw() = default;
+        Exception(const Exception& cause) throw() = default;
+        Exception& operator=(const Exception& e) throw() {
+            e.what();
```

Comment:
I don't see why you calling ``e.what()`` here because the return value is not being used. Also ``operator=`` is not really assigning anything

Reply:
Should I dip the whole file because I'm not using it anymore?

Reply:
yeah sure we can add exceptions if we need them in future